proposals/low-level-struct-improvements.md

```diff
@@ -366,13 +373,13 @@ Span<int> ComplexScopedRefExample(scoped ref Span<int> span)
 
 <a name="rules-method-arguments-must-match"></a>
 
-The presence of `ref` fields means the rules around method arguments must match need to be updated as a `ref` parameter can now be stored as a field in a `ref struct` argument to the method. Previously the rule only had to consider another `ref struct` being stored as a field. The impact of this is discussed in [the compat considerations](#compat-considerations). 
+The presence of `ref` fields means the rules around method arguments must match need to be updated as a `ref` parameter can now be stored as a field in a `ref struct` argument to the method. Previously the rule only had to consider another `ref struct` being stored as a field. The impact of this is discussed in [the compat considerations](#compat-considerations). The new rule is ... 
 
-The presence of `scoped` allows us to refine the rule to reduce the friction the rule creates. The `scoped` modifier lets us remove arguments from consideration as they cannot be returned from the method.
+> For any method invocation
+> 1. Calculate the *safe-to-escape* of the method return (for this rule assume it has a `ref struct` return)
```

Comment:
>assume it has a `ref struct` return [](http://example.com/codeflow?start=72&length=35)

Because we're treating a `ref struct` return value as a placeholder for any `ref` or `out` `ref struct` argument? If so, could we state that somewhere?

Reply:
It's because if it's a non-`ref struct` return then *safe-to-escape* of the return is trivially *calling method*. Effectively this is saying to "pretend" there is a `ref struct` return, calculate that *safe-to-escape* scope and then validate it can be assigned to every `ref` or `out` argument.